Source/CarthageKit/Xcode.swift

```diff
@@ -1192,7 +1201,16 @@ public func buildInDirectory(directoryURL: NSURL, withConfiguration configuratio
 			|> map { (scheme: String, project: ProjectLocator) -> BuildSchemeProducer in
 				let initialValue = (project, scheme)
 
-				let buildProgress = buildScheme(scheme, withConfiguration: configuration, inProject: project, workingDirectoryURL: directoryURL, sdkFilter: sdkFilter)
+				let wrappedSDKFilter: SDKFilterCallback = { sdks, scheme, configuration, project in
+					if platforms.isEmpty {
+						return sdkFilter(sdks: sdks, scheme: scheme, configuration: configuration, project: project)
+					}
+
+					let filteredSDKs = sdks.filter { platforms.contains($0.platform) }
+					return sdkFilter(sdks: filteredSDKs, scheme: scheme, configuration: configuration, project: project)
```

Comment:
I found this block a little confusing to read. Could we do it something like this instead?

```
let filteredSDKs: [SDK]
if platforms.isEmpty {
    filteredSDKs = sdks
} else {
    filteredSDKs = sdks.filter { ... }
}
return sdkFilter(...)
```

That makes it more obvious IMO.